psutil/_common.py

```diff
@@ -169,6 +171,39 @@ def supports_ipv6():
         return False
 
 
+def parse_environ_block(data):
+    """Parse a C environ block of environment variables into a dictionary."""
+    ret = {}
+    pos = 0
+
+    def convert_string(s):
+        if PY3 and not isinstance(data, bytes):
+            return s
+        else:
+            return s.decode(sys.getfilesystemencoding())
+
+    if PY3 and not isinstance(data, bytes):
+        nul, equals = '\0', '='
+    else:
+        nul, equals = b'\0', b'='
+
+    uppercase_key = os.name == "nt"
+
+    while True:
+        next_pos = data.find(nul, pos)
+        if next_pos <= pos:
+            break
+        equal_pos = data.find(equals, pos, next_pos)
+        if equal_pos > pos:
+            key = convert_string(data[pos:equal_pos])
+            value = convert_string(data[equal_pos+1:next_pos])
+            if uppercase_key:
+                key = key.upper()
+            ret[key] = value
+        pos = next_pos + 1
+    return ret
```

Comment:
I don't understand why this function is so complicated. You just need to split for '\0' and then for '='. This is an example: https://code.google.com/p/psutil/source/detail?r=3e6905e0c623#